src/main/java/frc/robot/subsystems/ElevatorSubsystem.java

```diff
@@ -0,0 +1,199 @@
+        m_elevatorISubs = ElevatorTuningTable.getDoubleTopic("I").getEntry(Config.ElevatorConfig.elevator_kI);
+        m_elevatorDSubs = ElevatorTuningTable.getDoubleTopic("D").getEntry(Config.ElevatorConfig.elevator_kD);
+        m_elevatorIzSubs = ElevatorTuningTable.getDoubleTopic("IZone").getEntry(Config.ElevatorConfig.elevator_kIz);
+        m_elevatorFFSubs = ElevatorTuningTable.getDoubleTopic("FF").getEntry(Config.ElevatorConfig.elevator_kFF);
+
+
+        m_elevatorFFSubs.setDefault(Config.ElevatorConfig.elevator_kFF);
+        m_elevatorPSubs.setDefault(Config.ElevatorConfig.elevator_kPDefault);
+        m_elevatorISubs.setDefault(Config.ElevatorConfig.elevator_kI);
+        m_elevatorDSubs.setDefault(Config.ElevatorConfig.elevator_kD);
+        m_elevatorIzSubs.setDefault(Config.ElevatorConfig.elevator_kIz);
+
+
+        // Send telemetry thru networktables
+        NetworkTable ElevatorDataTable = NetworkTableInstance.getDefault().getTable(m_dataTable);
+
+        m_targetPosition = ElevatorDataTable.getDoubleTopic("TargetPosition").publish(PubSubOption.periodic(0.02));
+        m_currentPosition = ElevatorDataTable.getDoubleTopic("CurrentPosition").publish(PubSubOption.periodic(0.02));
+
+
+        // PID config
+        m_elevator_config.closedLoop.feedbackSensor(ClosedLoopConfig.FeedbackSensor.kPrimaryEncoder)
+                .pid(m_elevatorPSubs.get(), m_elevatorISubs.get(), m_elevatorDSubs.get())
+                .velocityFF(0.003)
+                .outputRange(-1,1)
+                .maxMotion.maxVelocity(1000)
+                .maxAcceleration(1000)
+                .allowedClosedLoopError(0.25);
+
+        // configure elevator motor
+        m_elevator.configure(m_elevator_config, SparkBase.ResetMode.kResetSafeParameters, SparkBase.PersistMode.kPersistParameters);
+
+        m_elevator.setCANTimeout(0);
+
+        // reset encoder
+        m_elevator_encoder.setPosition(0);
+
+        ErrorTrackingSubsystem.getInstance().register(m_elevator);
+    }
+
+    @Override
+    public void periodic() {
+        // update networktables
+        m_currentPosition.accept(m_elevator_encoder.getPosition());
+        m_targetPosition.accept(elevatorTargetPos);
+
+        if (!controlOverride) {
+            // set elevator height
+            setElevatorHeight(elevatorTargetPos);
+        }
+    }
+
+    public void setElevatorHeight(double height) {
+        ClosedLoopSlot pidSlot = ClosedLoopSlot.kSlot0;
+        m_pidControllerElevator.setReference(height, ControlType.kPosition, pidSlot, 0);
+    }
+
+
+
+    //return positon
+    public double getPosition() {
+        return m_elevator.getEncoder().getPosition() - ElevatorConfig.shiftEncoderRange;
+    }
+
+
+    public double getVelocity() {
+        return m_elevator.getEncoder().getVelocity();
+    }
+
+
+    // Commands to manually set the motor.
+    public Command raiseMotorCommand() {
```

Comment:
We won't need this, since we are using position control. This will mess up the position.